### PR TITLE
feat: add request upgrade on endpoint config

### PR DESCRIPTION
**Pull request by gcusnieux**
## Issue

https://gravitee.atlassian.net/browse/APIM-2509

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-suegfnuttl.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-1-list.component.ts [hunk 1]
```diff
@@ -75,9 +75,20 @@ export class Step2Entrypoints1ListComponent implements OnInit, OnDestroy {
...
         this.changeDetectorRef.detectChanges();
       });
+
+    this.formGroup
+      .get('selectedEntrypointsIds')
+      .valueChanges.pipe(
+        tap((selectedEntrypointsIds) => {
+          this.shouldUpgrade = selectedEntrypointsIds
+            .map((id) => this.entrypoints.find((endpoint) => endpoint.id === id))
+            .some((endpoint) => !endpoint.deployed);
```
Comment (phiz71) [inline gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-1-list.component.ts]: Small typo
```suggestion
            .map((id) => this.entrypoints.find((entrypoint) => entrypoint.id === id))
            .some((entrypoint) => !entrypoint.deployed);
```

**Review by phiz71 — APPROVED** [review]
A small typo and a question, but LGTM.

File: gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-0-architecture.component.ts [hunk 2]
```diff
@@ -158,7 +158,7 @@ export class Step2Entrypoints0ArchitectureComponent implements OnInit, OnDestroy
     });
   }
 
-  public onRequestUpgrade($event: MouseEvent) {
-    this.licenseService.openDialog(this.licenseOptions, $event);
+  public onRequestUpgrade() {
+    this.licenseService.openDialog(this.licenseOptions);
```
Comment (phiz71) [inline gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-0-architecture.component.ts:162]: are you sure yo want to remove `$event`?

Comment (gcusnieux) [reply gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-0-architecture.component.ts:162]: Yes, it's an optionnal parameter. 
It's usefull when we doesn't want propagate the event but here there is no action/route under button, just the licence popup. 
For example, in the case of a menu item, when we want to block the access to the page

**Pull request merged**